contracts/token/ERC7984/extensions/ERC7984Rwa.sol

```diff
@@ -0,0 +1,242 @@
+    function _confidentialMint(address to, euint64 encryptedAmount) internal virtual returns (euint64) {
+        return _mint(to, encryptedAmount);
+    }
+
+    /// @dev Internal function which burns confidential amount of tokens from account.
+    function _confidentialBurn(address account, euint64 encryptedAmount) internal virtual returns (euint64) {
+        return _burn(account, encryptedAmount);
+    }
+
+    /// @dev Internal function which forces transfer of confidential amount of tokens from account to account by skipping compliance checks.
+    function _forceConfidentialTransferFrom(
+        address from,
+        address to,
+        euint64 encryptedAmount
+    ) internal virtual returns (euint64 transferred) {
+        if (!FHE.isInitialized(encryptedAmount)) {
+            return encryptedAmount;
+        }
+        encryptedAmount = FHE.select(
+            _preCheckForceTransfer(from, to, encryptedAmount),
+            encryptedAmount,
+            FHE.asEuint64(0)
+        );
+        // bypassing `from` restriction & frozen checks while `to` restriction check is still performed.
+        transferred = super._update(from, to, encryptedAmount); // bypass compliance check
+        _postForceTransfer(from, to, encryptedAmount);
+    }
+
+    /**
+     * @dev Bypasses the `from` restriction check when performing a {forceConfidentialTransferFrom}.
+     */
+    function _checkRestrictionFrom(address account) internal view override {
+        if (_isForceTransfer()) {
+            return;
+        }
+        super._checkRestrictionFrom(account);
+    }
+
+    /**
+     * @dev Bypasses the frozen check of the `from` account when performing a {forceConfidentialTransferFrom}.
+     */
+    function _getUnfrozenAvailableFrom(address account, euint64 encryptedAmount) internal override returns (euint64) {
+        if (_isForceTransfer()) {
+            return encryptedAmount;
+        }
+        return super._getUnfrozenAvailableFrom(account, encryptedAmount);
+    }
+
+    /// @dev Internal function which updates confidential balances while performing frozen, restriction and compliance checks.
+    function _update(
+        address from,
+        address to,
+        euint64 encryptedAmount
+    ) internal override(ERC7984Freezable, ERC7984Restricted, ERC7984) whenNotPaused returns (euint64 transferred) {
+        if (!FHE.isInitialized(encryptedAmount)) {
+            return encryptedAmount;
+        }
+        encryptedAmount = FHE.select(_preCheckTransfer(from, to, encryptedAmount), encryptedAmount, FHE.asEuint64(0));
+        // frozen and restriction checks performed through inheritance
+        transferred = super._update(from, to, encryptedAmount);
+        _postTransfer(from, to, encryptedAmount);
+    }
+
+    /// @dev Checks if a transfer follows compliance.
+    function _preCheckTransfer(address from, address to, euint64 encryptedAmount) internal virtual returns (ebool);
```

Comment:
should be `_preTransferCheck` or `_checkPreTransfer`

Reply:
Yes agree thank you, see [Rename to preTransferCheck](https://github.com/OpenZeppelin/openzeppelin-confidential-contracts/pull/160/commits/1de3990fad831e26d32c61805e8b5c625988aa8d).